sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSortMergeJoinExec.scala

```diff
@@ -64,13 +71,30 @@ class GpuSortMergeJoinMeta(
   }
 
   override def convertToGpu(): GpuExec = {
+    val buildSide = if (canBuildRight(join.joinType)) {
+      BuildRight
+    } else if (canBuildLeft(join.joinType)) {
+      BuildLeft
+    } else {
+      throw new IllegalStateException(s"Cannot build either side for ${join.joinType} join")
+    }
     GpuShuffledHashJoinExec(
       leftKeys.map(_.convertToGpu()),
       rightKeys.map(_.convertToGpu()),
       join.joinType,
-      BuildRight, // just hardcode one side
+      buildSide,
       condition.map(_.convertToGpu()),
       childPlans(0).convertIfNeeded(),
       childPlans(1).convertIfNeeded())
   }
+
+  private def canBuildRight(joinType: JoinType): Boolean = joinType match {
```

Comment:
might be nice to add comment about these must match the spark join selection critera for ShuffledHashJoinExec